discord-scripts/fix-twitter-embed.ts

```diff
@@ -62,27 +67,42 @@ async function workingTwitterEmbeds(
 //
 // See https://github.com/FixTweet/FxTwitter for more.
 export default function fixTwitterEmbeds(discordClient: Client, robot: Robot) {
+  // Process only messages that match the Twitter URL pattern
   discordClient.on("messageCreate", (message) => {
-    robot.logger.info(
+    robot.logger.debug(
       `fixTwitterEmbeds: processing new message ${message.content}`,
     )
-
-    workingTwitterEmbeds(message, robot.logger).catch((err) => {
-      robot.logger.error(
-        `fixTwitterEmbeds: failed to process new message ${message.content}: ${err}`,
+    if (message.content?.match(twitterUrlRegExp)) {
+      robot.logger.info(
+        `fixTwitterEmbeds: processing new message ${message.content}`,
       )
-    })
+      workingTwitterEmbeds(message, robot.logger).catch((err) => {
+        robot.logger.error(
+          `fixTwitterEmbeds: failed to process new message ${message.content}: ${err}`,
+        )
+      })
+    }
   })
 
   discordClient.on("messageUpdate", (oldMessage, newMessage) => {
-    robot.logger.info(
-      `fixTwitterEmbeds: processing updated message ${newMessage.content}, ${oldMessage.content}`,
+    robot.logger.debug(
+      `fixTwitterEmbeds: processing updated message ${newMessage.content}`,
     )
+    if (
+      newMessage.content?.match(twitterUrlRegExp) ||
+      oldMessage?.content?.match(twitterUrlRegExp)
+    ) {
+      robot.logger.info(
+        `fixTwitterEmbeds: processing updated message ${newMessage.content}`,
+      )
 
-    workingTwitterEmbeds(newMessage, robot.logger, oldMessage).catch((err) => {
-      robot.logger.error(
-        `fixTwitterEmbeds: failed to process new message ${newMessage.content}: ${err}`,
+      workingTwitterEmbeds(newMessage, robot.logger, oldMessage).catch(
+        (err) => {
+          robot.logger.error(
+            `fixTwitterEmbeds: failed to process updated message ${newMessage.content}: ${err}`,
```

Comment:
Same deal re: message content.

Actually I would also log the message id at info level, and only log the message content once alongside message id. That way we don't need to repeatedly spit the message content into the console.

Reply:
Just pushed a new commit with a bunch of refactoring as well hiding the whole content behind the `logger.debug` and then for `logger.info` we take user, channel, messageId and timestamp!